### ISSUE TITLE
Windows node stats failures

### ISSUE THREAD

**Issue by colinsurprenant** [elastic/logstash#8108]
These are specific node stats failures on Windows for `process.cpu.load_average.1m`. It seems this property is not available and I am not sure how we should address this both for reporting these values and testing them - I'd solicit the authors opinion on this? @ph @andrewvc ?

```
Failures:

  1) LogStash::Api::Modules::NodeStats GET / should set 'process.cpu.load_average.1m' at '/' to be a 'Numeric'
     ←[31mFailure/Error: expect(acc).to be_a(Hash), "Got a nil looking for #{resource_path} in #{payload}"←[0m
     ←[31m  Got a nil looking for ["process", "cpu", "load_average", "1m"] in {"host"=>"COLINSURPREF058", "version"=>"7.0.0-alpha1", "http_address"=>"127.0.0.1:9600", "id"=>"e1db7b7a-3975-4788-8721-9c
6b0f7c7774", "name"=>"COLINSURPREF058", "jvm"=>{"threads"=>{"count"=>8, "peak_count"=>10}, "mem"=>{"heap_used_percent"=>9, "heap_committed_in_bytes"=>201613312, "heap_max_in_bytes"=>1056309248, "heap_
used_in_bytes"=>98970112, "non_heap_used_in_bytes"=>89888432, "non_heap_committed_in_bytes"=>96952320, "pools"=>{"survivor"=>{"peak_used_in_bytes"=>2228224, "used_in_bytes"=>2228224, "peak_max_in_byte
s"=>17432576, "max_in_bytes"=>17432576, "committed_in_bytes"=>2228224}, "old"=>{"peak_used_in_bytes"=>163232208, "used_in_bytes"=>81277128, "peak_max_in_bytes"=>899284992, "max_in_bytes"=>899284992, "
committed_in_bytes"=>181493760}, "young"=>{"peak_used_in_bytes"=>17891328, "used_in_bytes"=>15464760, "peak_max_in_bytes"=>139591680, "max_in_bytes"=>139591680, "committed_in_bytes"=>17891328}}}, "gc"
=>{"collectors"=>{"old"=>{"collection_time_in_millis"=>88, "collection_count"=>5}, "young"=>{"collection_time_in_millis"=>1167, "collection_count"=>126}}}, "uptime_in_millis"=>21234}, "process"=>{"ope
n_file_descriptors"=>-1, "peak_open_file_descriptors"=>-1, "max_file_descriptors"=>-1, "mem"=>{"total_virtual_in_bytes"=>-1}, "cpu"=>{"total_in_millis"=>-1, "percent"=>-3, "load_average"=>nil}}, "even
ts"=>{"in"=>100, "filtered"=>100, "out"=>100, "duration_in_millis"=>140, "queue_push_duration_in_millis"=>0}, "pipelines"=>{"main"=>{"events"=>{"duration_in_millis"=>140, "in"=>100, "out"=>100, "filte
red"=>100, "queue_push_duration_in_millis"=>0}, "plugins"=>{"inputs"=>[{"id"=>"api-generator-pipeline", "events"=>{"out"=>100, "queue_push_duration_in_millis"=>0}, "name"=>"generator"}], "filters"=>[]
, "outputs"=>[{"id"=>"5d2e9c9b7c6171771d1f9510313fda837d1a888b83d531fbc9512e3a929dab1a", "events"=>{"duration_in_millis"=>0, "in"=>100, "out"=>100}, "name"=>"dummyoutput"}]}, "reloads"=>{"last_error"=
>nil, "successes"=>0, "last_success_timestamp"=>nil, "last_failure_timestamp"=>nil, "failures"=>0}, "queue"=>{"type"=>"memory"}}}, "reloads"=>{"successes"=>0, "failures"=>0}, "os"=>{}}←[0m
     ←[36m# ./spec/support/resource_dsl_methods.rb:66:in `block in test_api'←[0m
     ←[36m# ./spec/support/resource_dsl_methods.rb:65:in `block in test_api'←[0m
     ←[36m# ./vendor/bundle/jruby/2.3.0/gems/rspec-wait-0.0.9/lib/rspec/wait.rb:46:in `block in (root)'←[0m
     ←[36m# C:\Users\colin\dev\logstash\lib\bootstrap\rspec.rb:13:in `<main>'←[0m

  2) LogStash::Api::Modules::NodeStats GET /process should set 'process.cpu.load_average.1m' at '/process' to be a 'Numeric'
     ←[31mFailure/Error: expect(acc).to be_a(Hash), "Got a nil looking for #{resource_path} in #{payload}"←[0m
     ←[31m  Got a nil looking for ["process", "cpu", "load_average", "1m"] in {"host"=>"COLINSURPREF058", "version"=>"7.0.0-alpha1", "http_address"=>"127.0.0.1:9600", "id"=>"e1db7b7a-3975-4788-8721-9c
6b0f7c7774", "name"=>"COLINSURPREF058", "process"=>{"open_file_descriptors"=>-1, "peak_open_file_descriptors"=>-1, "max_file_descriptors"=>-1, "mem"=>{"total_virtual_in_bytes"=>-1}, "cpu"=>{"total_in_
millis"=>-1, "percent"=>-3, "load_average"=>nil}}}←[0m
     ←[36m# ./spec/support/resource_dsl_methods.rb:66:in `block in test_api'←[0m
     ←[36m# ./spec/support/resource_dsl_methods.rb:65:in `block in test_api'←[0m
     ←[36m# ./vendor/bundle/jruby/2.3.0/gems/rspec-wait-0.0.9/lib/rspec/wait.rb:46:in `block in (root)'←[0m
     ←[36m# C:\Users\colin\dev\logstash\lib\bootstrap\rspec.rb:13:in `<main>'←[0m

Finished in 3.03 seconds (files took 5.5 seconds to load)
←[31m109 examples, 2 failures←[0m

Failed examples:

←[31mrspec './logstash-core/spec/logstash/api/modules/node_stats_spec.rb[1:2:34]'←[0m ←[36m# LogStash::Api::Modules::NodeStats GET / should set 'process.cpu.load_average.1m' at '/' to be a 'Numeric'←[
0m
←[31mrspec './logstash-core/spec/logstash/api/modules/node_stats_spec.rb[1:4:9]'←[0m ←[36m# LogStash::Api::Modules::NodeStats GET /process should set 'process.cpu.load_average.1m' at '/process' to be
a 'Numeric'←[0m

Randomized with seed 58438
```

**Comment by ph** [comment]
@colinsurprenant we are using the mxbean to retrieve that information, theses beans are platform dependent and will not return the load average on windows.

This is consistent with elasticsearch (6.0.0) of the node stats api. http://localhost:9200/_nodes/stats:

**Linux**
```
cpu: {
percent: 26,
load_average: {
1m: 3.18798828125
}
```

**Windows**
```
cpu: {
percent: 36,

}
```

Could we make this test platform dependent?

**Comment by colinsurprenant** [comment]
@ph yes we can - but I just want to make sure that there isn't more concerns than just having a Windows specific test that will ignore this property? Do we need to make other changes in the stats api? can this lead to other concerns downstream where the stats are used, x-pack? visualisations?

I'd like folks closer to these concerns to followup on that, once settled I can help in modifying the tests accordingly?

**Comment by suyograo** [comment]
Let's make this test platform dependent. I also confirmed ES doesn't report this stats for windows as well, so we can discuss an alternate strategy along with them for the longer term. 

As for x-pack visualization, it's the same. We don't show it when LS is running on Windows. We can handle it gracefully in the UI (i think there is an issue already).

**Comment by andrewvc** [comment]
+1 on making it platform dependent.

**Comment by colinsurprenant** [comment]
ok fixe included in #8116

**Comment by colinsurprenant** [comment]
closing, #8116 merged. I will trust @ph @suyograo @andrewvc  to followup for any downstream consequences, if any, of not reporting the `process.cpu.load_average.1m` attribute on Windows.
